orc8r/gateway/python/magma/magmad/metrics.py

```diff
@@ -287,13 +287,18 @@ def _collect_service_metrics():
         memory_limit = properties[3]
 
         if pid != 0:
-            p = psutil.Process(pid=pid)
-            cpu_percentage = p.cpu_percent(interval=1)
-            _counter_set(
-                SERVICE_CPU_PERCENTAGE.labels(
-                    service_name=service,
-                ), cpu_percentage,
-            )
+            try:
+                p = psutil.Process(pid=pid)
+                cpu_percentage = p.cpu_percent(interval=1)
+            except psutil.NoSuchProcess:
+                logging.warning("When collecting CPU usage for service %s: Process with PID %d no longer exists.", service, pid)
+                continue
+            else:
```

Comment:
This indentation seems a little off... what's this "else" corresponding with?

Reply:
The `else` is aligned correctly with the `try` block here and will only be executed if the `except` statement is not executed.

Reply:
TIL 🙂